ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -108,10 +112,57 @@ prop_Mempool_InvalidTxsNeverAdded bc txs =
     genTxIsInvalid (TestGenTx (InvalidTestTx _)) = True
     genTxIsInvalid _                             = False
 
+-- | Test that all valid transactions added to a 'Mempool' via 'addTxs' are
+-- appropriately represented in the trace of events.
+prop_Mempool_TraceValidTxs :: Chain TestBlock
+                           -> [TestTx]
+                           -> Property
+prop_Mempool_TraceValidTxs bc txs =
+    testAddTxsWithTrace bc txs traceProp
+  where
+    traceProp :: [GenTx TestBlock] -> [TraceEventMempool TestBlock] -> Property
+    traceProp genTxs es = case (filter isAddTxsEvent es) of
+      [TraceMempoolAddTxs addedTxs _] ->
+        sortGenTxs (filter genTxIsValid genTxs) === sortGenTxs addedTxs
+      _ -> es === [] .&&. (filter genTxIsValid genTxs) === []
+    --
+    genTxIsValid :: GenTx TestBlock -> Bool
+    genTxIsValid (TestGenTx (ValidTestTx _)) = True
+    genTxIsValid _                           = False
+    --
+    isAddTxsEvent :: TraceEventMempool blk -> Bool
+    isAddTxsEvent (TraceMempoolAddTxs _ _) = True
+    isAddTxsEvent _                        = False
+
+-- | Test that all invalid rejected transactions returned from 'addTxs' are
+-- appropriately represented in the trace of events.
+prop_Mempool_TraceRejectedTxs :: Chain TestBlock
+                              -> [TestTx]
+                              -> Property
+prop_Mempool_TraceRejectedTxs bc txs =
+    testAddTxsWithTrace bc txs traceProp
+  where
+    traceProp :: [GenTx TestBlock] -> [TraceEventMempool TestBlock] -> Property
+    traceProp genTxs es = case (filter isRejectedTxsEvent es) of
+      [TraceMempoolRejectedTxs rejectedTxs _] ->
+        sortGenTxs (filter genTxIsInvalid genTxs) === sortGenTxs rejectedTxs
+      _ -> es === [] .&&. (filter genTxIsInvalid genTxs) === []
+    --
+    genTxIsInvalid :: GenTx TestBlock -> Bool
+    genTxIsInvalid (TestGenTx (InvalidTestTx _)) = True
+    genTxIsInvalid _                             = False
+    --
+    isRejectedTxsEvent :: TraceEventMempool blk -> Bool
+    isRejectedTxsEvent (TraceMempoolRejectedTxs _ _) = True
+    isRejectedTxsEvent _                             = False
+
 {-------------------------------------------------------------------------------
   Helpers
 -------------------------------------------------------------------------------}
 
+sortGenTxs :: [GenTx TestBlock] -> [GenTx TestBlock]
```

Comment:
If we derive an `Ord` instance for `GenTx TestBlock`, we can simply use `sort`:
```haskell
instance ApplyTx TestBlock where
  newtype GenTx TestBlock = TestGenTx
    { testGenTx :: TestTx
    } deriving (Show, Eq, Ord)
```

Reply:
:man_facepalming: